python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -605,3 +622,298 @@ def makeDataRefList(self, namespace):
+        - for big areas do not use the region as determined from the coadd, but rather all overlapping
+          footprints from the visit DETECTED plane.  This helps in flagging low surface brightness regions.
+        - assemble the coadd, ignoring the clipped regions.
+
+    The cuts used to determine the clipped regions were found empirically by examining a lot of HSC data.
+    It is unlikely that they will generalize well to other data sets.  The current algorithm is limited to
+    identifying artifacts that are found in only one or two visits.  A different threshold is applied
+    depending on the number overlapping visits.  The current thresholds work well at removing a majority
+    of artifacts.
+
+    """
+    ConfigClass = SafeClipAssembleCoaddConfig
+    _DefaultName = "safeClipAssembleCoadd"
+
+    def __init__(self, *args, **kwargs):
+        super(SafeClipAssembleCoaddTask, self).__init__(*args, **kwargs)
+        schema = afwTable.SourceTable.makeMinimalSchema()
+        self.makeSubtask("clipDetection", schema=schema)
+
+    def assemble(self, skyInfo, tempExpRefList, imageScalerList, weightList, bgModelList):
+        """Assemble the coadd for a region
+
+        Identify clipped regions by detecting objects on the difference between unclipped and clipped coadd
+        and then flag these regions on the individual vists so they are ignored in the coaddition process
+
+        @param skyInfo: Patch geometry information, from getSkyInfo
+        @param tempExpRefList: List of data reference to tempExp
+        @param imageScalerList: List of image scalers
+        @param weightList: List of weights
+        @param bgModelList: List of background models from background matching
+        return coadd exposure
+        """
+        exp = self.buildDifferenceImage(skyInfo, tempExpRefList, imageScalerList, weightList, bgModelList)
+        mask = exp.getMaskedImage().getMask()
+        mask.addMaskPlane("CLIPPED")
+
+        result = self.detectClip(exp, tempExpRefList)
+
+        self.log.info('Found %d clipped objects' % len(result.clipFootprints))
+
+        # Go to individual visits for big footprints
+        maskClipValue = mask.getPlaneBitMask("CLIPPED")
+        maskDetValue = mask.getPlaneBitMask("DETECTED") | mask.getPlaneBitMask("DETECTED_NEGATIVE")
+        bigFootprints = self.detectClipBig(result.tempExpClipList, result.clipFootprints, result.clipIndices,
+                                           maskClipValue, maskDetValue)
+
+        # Create mask of the current clipped footprints
+        maskClip = mask.Factory(mask.getBBox(afwImage.PARENT))
+        afwDet.setMaskFromFootprintList(maskClip, result.clipFootprints, maskClipValue)
+
+        maskClipBig = maskClip.Factory(mask.getBBox(afwImage.PARENT))
+        afwDet.setMaskFromFootprintList(maskClipBig, bigFootprints, maskClipValue)
+        maskClip |= maskClipBig
+
+        # Now combine again using mean, ignoring CLIPPED pixels
```

Comment:
"Combine again using mean" could be more helpful. Specifically, I think you mean you are passing `doClip=False` to `AssembleCoaddTask.assemble()` a few lines down.